WeatherStationClient.py

```diff
@@ -0,0 +1,258 @@
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+        
+    #TODO: Czy poprawna składnia URI jeśli nie ma filter
+    async def station_transmission_history_last(self, station_id, amount, filter_=None, sort=None):
+        """Read last X amount of station transmission history. Optionally you can also sort them ASC or DESC and filter."""
+        uri = f'station/{station_id}/history'
+        if filter_ is not None:
+            uri = uri + f'/{filter_}'
+        uri = uri + f'/{last}/{amount}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_transmission_history_between(self, station_id, from_unix_timestamp, to_unix_timestamp, filter_=None, sort=None):
+        """Read transmission history for specific time period. Optionally you can also sort them ASC or DESC and filter."""
+        uri = f'station/{station_id}/history/'
+        if filter_ is not None:
+            uri = uri + f'/{filter_}'
+        uri = uri + f'/from/{from_unix_timestamp}/to/{to_unix_timestamp}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_licenses(self, station_id):
+        """Retrieve all the licenses of your device. They are separated by the service (models, forecast ...)."""
+        return await self._send('GET', f'station/{station_id}/licenses')
+    
+    
+
+class ClientBuilder:
+
+    class _RequestContents:
+        def __init__(self, method, route, body, headers):
+            self.method = method
+            self.route = route
+            self.headers = headers
+            self.body = body
+    
+    class _ConnectionBase:
+        apiURI = 'https://api.fieldclimate.com/v1'
+    
+        async def __aenter__(self):
+            self._session = aiohttp.ClientSession()
+            return ApiClient(self)
+        
+        async def __aexit__(self, exc_type, exc_value, traceback):
+            await self._session.close()
+            
+        @abstractmethod
+        def _modifyRequest(self, requestContents):
+            pass
+        
+        async def _make_request(self, method, route, body=None):
+            requestContents = ClientBuilder._RequestContents(method, route, body, {'Accept': 'application/json'})
+            self._modifyRequest(requestContents)
+            
+            args = [self.apiURI+'/'+requestContents.route]
+            kwargs = {'headers': requestContents.headers}
+            if requestContents.body is not None:
+                kwargs['json'] = body
+            reqMethod = {
+                'GET': self._session.get,
+                'PUT': self._session.put,
+                'DELETE': self._session.delete,
+                'POST': self._session.post
+            }[method]
+            
+            result = await reqMethod(*args, **kwargs)
+            if result.status == 400:
```

Comment:
As mentioned before we could use build in exception by ie.:
`self._session = aiohttp.ClientSession(raise_for_status=True)`